state/dbstore.go

```diff
@@ -105,6 +106,18 @@ func (s *DBStore) Delete(key string) (err error) {
 	return s.db.Delete([]byte(key), nil)
 }
 
+// Keys returns a list of all the keys in the underlying LevelDB.
+func (s *DBStore) Keys() (keys []string, err error) {
+	iter := s.db.NewIterator(nil, nil)
```

Comment:
you could just after this line say
`defer iter.Release` and then at the end just

`return keys, iter.Error()`

Reply:
Done, nice catch!

I've also updated the code so that either `keys` or `err` _have_ to be `nil`.